mirage/resolver/dns_resolver_mirage_shared.mli

```diff
@@ -19,4 +19,7 @@ module type S = sig
   (** [update_tls t tls_config] updates the tls configuration to [tls_config].
       If the resolver wasn't already listening for TLS connections it will
       start listening. *)
+
+  val queries : t -> (Ptime.t * Dns.Packet.Question.t * Ipaddr.t * Dns.Rcode.t * int64 * string) Lwt_condition.t
```

Comment:
Given that this is exported I think we should use a record so we have meaningful names to each field. For example it's not clear what the `int64` and the `string` is.